SwrveSDK/src/amazon/java/com/swrve/sdk/SwrvePushEngageReceiver.java

```diff
@@ -0,0 +1,112 @@
+package com.swrve.sdk;
+
+import android.app.PendingIntent;
+import android.content.BroadcastReceiver;
+import android.content.Context;
+import android.content.Intent;
+import android.os.Bundle;
+
+import com.swrve.sdk.adm.SwrveAdmConstants;
+import com.swrve.sdk.adm.SwrveAdmNotification;
+
+import org.json.JSONException;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+public class SwrvePushEngageReceiver extends BroadcastReceiver {
+    private static final String LOG_TAG = "SwrveAdm";
+
+    private Context context;
```

Comment:
The broadcast receiver will always be active, it would be better to not save these references, only use them in onReceive etc.